client/core/wallet.go

```diff
@@ -106,8 +122,7 @@ func (w *xcWallet) unlocked() bool {
 // this is true only if the decrypted password is cached. Use this to determine
 // if the wallet may be unlocked without user interaction (via refreshUnlock).
 func (w *xcWallet) locallyUnlocked() bool {
-	// return len(w.encPW) == 0 || len(w.pw) > 0
-	if len(w.encPW) == 0 {
+	if len(w.encPW()) == 0 {
```

Comment:
In both returns, the `w.mtx` gets locked one way or the other, so this should be:

```go
w.mtx.RLock()
defer w.mtx.RUnlock()
if len(w.encPass) == 0 {
    return true // unencrypted wallet
}
return len(w.pw) > 0 // cached password for encrypted wallet
```